### DISCUSSION TITLE
GraphQL: Unable to match type definition (TypeName{name='Long'})

### DISCUSSION THREAD

**Discussion by rajarshp** [graphql-java-kickstart/graphql-spring-boot#667] [Q&A]
Hello,
I am just learning graphql using Springboot and was trying some simple programs. Now I have encountered a problem and not able to understand why it's throwing this error.

Can someone please help me to point out what mistake I have made or is it a Bug?

**Error:** `graphql.kickstart.tools.SchemaClassScannerError: Unable to match type definition (TypeName{name='Long'}) with java type (class java.lang.Long): No TypeDefinition for type name Long`

**author.graphqls**

```
type Author {
    id: Long!
    name: String!
    age: Int
}

# Root
type Query {
    findAuthorById(id: Long!): Author!
    findAllAuthors: [Author]!
    countAuthors: Long!
}

# Root
type Mutation {
    createAuthor(name: String!, age: Int): Author!
}
```
**Author.java**

```
@Entity
public class Author {
  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  private Long id;

  @Column(name = "name", nullable = false)
  private String name;

  @Column(name = "age")
  private Integer age;
```

**Comment by abhi10005** [comment]
Long data type is not supported as default available type in graphql. However, you can use extended scalar lib (graphql-java-extended-scalars) to use types other than default ones provided by graphql.

Once you have this dependency in your pom.xml or build.gradle, you need to define a bean for it as follows (assuming you are using Spring):

```
@Configuration
public class ScalarConfig {

    @Bean public GraphQLScalarType graphQLLong() {
        return ExtendedScalars.GraphQLLong;
    }
}

```
Then, declare this type in your graphqls schema as follows:
`scalar Long`

You will be good to go!
